pkg/virt-api/rest/subresource.go

```diff
@@ -307,20 +313,12 @@ func (app *SubresourceAPIApp) StartVMRequestHandler(request *restful.Request, re
 		patchType = types.JSONPatchType
 
 		needsRestart := false
-		vmi, err := app.VirtCli.VirtualMachineInstance(namespace).Get(name, &k8smetav1.GetOptions{})
-		if err != nil {
-			if !errors.IsNotFound(err) {
-				response.WriteError(http.StatusInternalServerError, err)
-				return
-			}
-		} else {
-			if (runStrategy == v1.RunStrategyRerunOnFailure && vmi.Status.Phase == v1.Succeeded) ||
-				(runStrategy == v1.RunStrategyManual && vmi.IsFinal()) {
-				needsRestart = true
-			} else if runStrategy == v1.RunStrategyRerunOnFailure && vmi.Status.Phase == v1.Failed {
-				response.WriteError(http.StatusBadRequest, fmt.Errorf("runstrategy rerunonerror does not support starting VM from failed state"))
-				return
-			}
+		if (runStrategy == v1.RunStrategyRerunOnFailure && vmi.Status.Phase == v1.Succeeded) ||
```

Comment:
I think you need to check `vmi != nil` here. In the earlier version this was inside an err == nil block, which means that there also was no notFound error, and so `vmi` had a value for sure

Reply:
Yeah, actually I find the nil checks already a little awkward. Will improve this.